pipelines/coref/project.yml

```diff
@@ -0,0 +1,127 @@
+title: "spaCy Coref Model"
+description: >
+  This project trains a coreference model for spaCy using OntoNotes.
+
+vars:
+  # XXX Change to your actual GPU ID
+  gpu_id: 0
+  # XXX fill with your local path to OntoNotes
+  ontonotes: /home/polm/data/ontonotes
+  max_epochs: 5
+
+directories: ["assets", "corpus", "scripts", "configs", "training"]
+
+assets:
+  - dest: "assets/"
+    git:
+      repo: "https://github.com/explosion/conll-2012"
+      branch: "main"
+      path: ""
+    description: "CoNLL-2012 scripts and dehydrated data."
+  - dest: "assets/conll-eval"
+    git:
+      repo: "https://github.com/conll/reference-coreference-scorers"
+      branch: "master"
+      path: ""
+    description: "CoNLL-2012 reference scorer."
+
+workflows:
+  prep:
+    - prep-data
+    - preprocess
+  train:
+    - train-cluster
+    - prep-span-data
+    - train-span-predictor
+    - assemble
```

Comment:
This is really weird, but I ran `spacy project run train` today and it ran `train-cluster`, trained the model, saved it, and then quit. `prep-span-data` etc weren't executed?

```
(venv_spacy)
Sofie@DESKTOP-NATQ3P4 MINGW64 .../projects/pipelines/coref (feature/coref)
$ spacy project run train
ℹ Running workflow 'train'

=============================== train-cluster ===============================
Running command: '...python.exe' -m spacy train configs/cluster.cfg -g 0 --paths.train corpus/train.spacy --paths.dev corpus/dev.spacy -o training/cluster --training.max_epochs 5
ℹ Saving to output directory: training\cluster
ℹ Using GPU: 0

=========================== Initializing pipeline ===========================
[2022-04-29 11:58:25,012] [INFO] Set up nlp object from config
...
✔ Initialized pipeline

============================= Training pipeline =============================
ℹ Pipeline: ['sentencizer', 'transformer', 'coref']
ℹ Set annotations on update for: ['sentencizer']
ℹ Initial learn rate: 0.0001
E    #       LOSS TRANS...  LOSS COREF  SENTS_F  COREF_F  COREF_P  COREF_R  SCORE
---  ------  -------------  ----------  -------  -------  -------  -------  ------
  0       0           0.00      204.01    83.20     0.00     0.00     0.00    0.00
  0    1000           0.00    36042.83    83.20    35.34    52.41    26.65    0.35
...
  2    7000           0.00    22403.98    83.20    56.39    73.86    45.61    0.56
✔ Saved pipeline to output directory
training\cluster\model-last

(venv_spacy)
Sofie@DESKTOP-NATQ3P4 MINGW64 .../projects/pipelines/coref (feature/coref)
```

Reply:
That is really weird... Could it be Windows related? 

I assume the training must have exited with a non-zero code but I can't imagine why if the model was saved.

Reply:
Can you let me know if this still happens? I have been keeping an eye out, but I have never seen it happen and I am not sure what could cause it. If it does happen I guess I'd want to check the exit code of the process.

Reply:
Trying this again on a fresh system today!

Reply:
Unfortunately I couldn't replicate or refute this on the fresh system as it ran out of CUDA memory during training (which was sort of expected, it's only a laptop) and I don't have access to my normal/old/proper desktop system for at least another few days.

Reply:
I suggest we keep this in the back of our mind though, in case we see reports of something similar once we release this. I'll resolve in the meantime.

Reply:
Ugh, unfortunately this happened again.
```
$ spacy project run train
(...)
  6   19000           0.00    16163.36    67.59    69.78    65.54    0.68
  7   20000           0.00    15567.39    67.79    71.76    64.24    0.68
  7   21000           0.00    16047.34    67.66    69.10    66.28    0.68
  7   22000           0.00    15536.11    67.47    69.54    65.52    0.67
✔ Saved pipeline to output directory
training\cluster\model-last
``` 
The script did save the models, but then just exited without running the remainders of the commands. I suspect there's some reason why it crashes, but this error gets masked by the project system somehow.

Reply:
Ok, did a few more tests, and it looks like my hunch was correct: there was not enough space on my system to safe the very final model, but this error didn't make it to the console somehow. When I try again with smaller data/less iterations, it "trains" the model, then does continue to `prep-span-data`. So this is probably more of a UX issue with `spacy project` or maybe even with `spacy train` then it is an issue with this specific project.